shopify-products-with-openai/creatproducts/service.bal

```diff
@@ -0,0 +1,156 @@
+import ballerinax/shopify.admin as shopify;
+import ballerinax/googleapis.sheets;
+import ballerina/http;
+// import ballerina/io;
+
+// Google Sheets API client configuration
+configurable string gsheetToken = ?;
+configurable string gsheetsClientID = ?;
+configurable string gsheetsClientSecret = ?;
+configurable string gsheetsRefreshToken = ?;
+
+// OpenAI token
+configurable string openAIToken = ?;
+
+// Shopify token 
+configurable string shopifyToken = ?;
+configurable string shopifyStoreURL = ?;
+
+// Google Sheets client configuration
+sheets:Client sheetsEp = check new (config = {
+    auth: {
+        refreshUrl: sheets:REFRESH_URL,
+        clientId: gsheetsClientID,
+        clientSecret: gsheetsClientSecret,
+        refreshToken: gsheetsRefreshToken
+    }
+});
+
+// OpenAI API client configuration
+http:Client openAI = check new ("https://api.openai.com");
+
+// Shopify client configuration
+shopify:Client shopifyEp = check new (apiKeyConfig = {
+    xShopifyAccessToken: shopifyToken
+}, serviceUrl = shopifyStoreURL);
+
+// Product record
+type Product record {
+    string name;
+    string benefits;
+    string features;
+    string usedBy;
+    string price;
+    string productType;
+};
+
+type OpenAIImagePrompt record {
+    string prompt;
+    int n = 1;
+    string size = "256x256";
+};
+
+// OpenAI Prompt record
+type OpenAICompletionPrompt record {
+    string model = "text-davinci-003";
+    string prompt;
+    float temperature = 0.7;
+    int max_tokens = 256;
+    float top_p = 1;
+    int frequency_penalty = 0;
+    int presence_penalty = 0;
+};
+
+type OpenAICompletionChoicesItem record {
+    string text;
+    int index;
+    anydata logprobs;
+    string finish_reason;
+};
+
+type OpenAICompletionUsage record {
+    int prompt_tokens;
+    int completion_tokens;
+    int total_tokens;
+};
+
+type OpenAICompletionResponse record {
+    string id;
+    string 'object;
+    int created;
+    string model;
+    OpenAICompletionChoicesItem[] choices;
+    OpenAICompletionUsage usage;
```

Comment:
Seems we can keep only `choices` (as it Is the interested value) and remove other fields as this is an open record.